parsel/csstranslator.py

```diff
@@ -89,6 +101,116 @@ def xpath_text_simple_pseudo_element(self, xpath):
         """Support selecting text nodes using ::text pseudo-element"""
         return XPathExpr.from_xpath(xpath, textnode=True)
 
+    def xpath_first_of_pseudo(self, xpath):
```

Comment:
`div:nth-of-type(2)` would be the same as `div:nth-of(2)` (I haven't tested, but probably should)
the difference is when you have classes before `:nth-of`

`div.content.green:nth-of-type(2)` vs. `div.content.green:nth-of(2)`, the latter selecting the 2nd div matching ".green" and ".content" (the former matching only div that are 2nd child, and also .green and .content)